StacMan/ReflectionCache.cs

```diff
@@ -1,28 +1,10 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Linq;
-using System.Reflection;
 
 namespace StackExchange.StacMan
 {
     internal static class ReflectionCache
     {
-        public static readonly MethodInfo StacManClientParseApiResponse = typeof(StacManClient).GetMethod("ParseApiResponse", BindingFlags.NonPublic | BindingFlags.Instance);
-
-        public static class ApiFieldsByName<T> where T : StacManType
-        {
-            static ApiFieldsByName()
-            {
-                Value = typeof(T).GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                    .Where(pi => pi.GetCustomAttributes(typeof(FieldAttribute), false).Length == 1)
-                    .ToDictionary(
-                        pi => ((FieldAttribute)pi.GetCustomAttributes(typeof(FieldAttribute), false)[0]).FieldName,
-                        pi => pi);
-            }
-
-            public static readonly IDictionary<string, PropertyInfo> Value;
-        }
-
         public static class SortsBySortType<TSort> where TSort : struct // proxy for "where TSort: enum"
         {
             static SortsBySortType()
```

Comment:
> This is not used.

I don't understand what you mean, I think it's definitely used. Maybe you got fooled by all the parameters being grayed out by ReSharper? (They are greyed out because the are only used for precondition checks, which is the purpose of this `ValidateSortMinMax` method.)

> The same question for `ApiFieldsByName`.

I haven't really looked into this. I saw some manual code to map the JSON with `System.Web.Script.Serialization` (which I have never used) and figured that replacing all this manual mapping with `System.Text.Json` would be semantically equivalent. All the tests passing comforted me into this idea.

Reply:
ad `SortsBySortType` - please disregard my remark. I must have overlooked that
`ApiFieldsByName` - yeah, that seems to be the case :)

let's see what @emmettnicholas has to say regarding this PR